### ISSUE TITLE
Wayland fullscreen does not use screen priority

### ISSUE THREAD

**Issue by CorneliusCornbread** [redeclipse/base#1563]
- Version and build: [2.0.9] [steam:723]
- Observation: Upon attempting to set the game to fullscreen it will freak out between my main and right vertical monitor and then display on only my vertical monitor, stretched.
- Expected result: I should be either able to select which monitor the fullscreen uses or it should follow the priority of my monitors set in KDE plasma
- How to reproduce: Open game in a Wayland compatible desktop, KDE is what I used. Fullscreen game, exclusive or borderless, watch freakout happen

```
Insert the contents of your `log.txt` here.
```
I cannot find for the life of me where the log.txt output is. It'd be nice if it was documented somewhere like the readme.

**Comment by CorneliusCornbread** [comment]
Forcing the game to use the X11 backend `SDL_VIDEODRIVER=x11 %command%` and using XWayland it works just fine.

**Comment by qreeves** [comment]
Yeah I haven't tested under Wayland yet, I daresay it's the fail-safe routines for retina display detection kicking in it something. 

The log should be in `~/.redeclipse`

**Comment by CorneliusCornbread** [comment]
Here's my log, launching the game _with_ fullscreen enabled still leads to it full screening on the wrong monitor but it's a little more well behaved as it now places it just on my left-most monitor unstretched.
[log.txt](https://github.com/user-attachments/files/16132854/log.txt)

**Comment by qreeves** [comment]
Awesome, thanks. Will take a look soon.

**Comment by user1-github** [comment]
@CorneliusCornbread So you tried running the game with SDL_VIDEODRIVER=wayland?

Because if I set SDL_VIDEODRIVER=wayland in Steam launch options, the game just doesn't launch. My setup is slightly different: I have a single monitor, Gnome Wayland and I use Steam Flatpak.

**Comment by qreeves** [comment]
I use Wayland myself and have observed a number of issues. I will get to examining the problem in the near future.

**Comment by CorneliusCornbread** [comment]
> @CorneliusCornbread So you tried running the game with SDL_VIDEODRIVER=wayland?
> 
> Because if I set SDL_VIDEODRIVER=wayland in Steam launch options, the game just doesn't launch. My setup is slightly different: I have a single monitor, Gnome Wayland and I use Steam Flatpak.

That's likely an issue with Steam Flatpak, not redeclipse. Flatpak steam has issues with Wayland.

**Comment by user1-github** [comment]
> That's likely an issue with Steam Flatpak, not redeclipse. Flatpak steam has issues with Wayland.

Seems so. I reported it to Steam Flatpak maintainers.
